### ISSUE TITLE
Cannot connect to the Internet

### ISSUE THREAD

**Issue by angad9** [jade-hpc-gpu/jade-hpc-gpu.github.io#48]
A simple ping google.com is not working. Moreover, we are being unable to commit to git or install libraries to setup the python environment.

**Comment by we-taper** [comment]
Me too, the usual `git pull` now returns
```
Received HTTP code 503 from proxy after CONNECT
```

**Comment by andygittings** [comment]
I am seeing this too:

git clone https://github.com/pydicom/pydicom.git
Cloning into 'pydicom'...
fatal: unable to access 'https://github.com/pydicom/pydicom.git/': Received HTTP code 503 from proxy after CONNECT


$ wget https://github.com/pydicom/pydicom/archive/master.zip
--2018-05-04 07:48:29--  https://github.com/pydicom/pydicom/archive/master.zip
Resolving xcat (xcat)... 172.31.124.23
Connecting to xcat (xcat)|172.31.124.23|:3128... connected.
Proxy tunneling failed: Service UnavailableUnable to establish SSL connection.

**Comment by LiamATOS** [comment]
Hi, 

What Node are you trying to do this from? Login or compute?

Thanks

Liam

**Comment by andygittings** [comment]
Hi Liam,

Login nodes...

Cheers,

Andy

**Comment by LiamATOS** [comment]
Hi, 

Looks resolv.conf was beign over written for the login nodes, can you try again please?

Thanks

Liam

**Comment by angad9** [comment]
It still doesn't work for me :(

**Comment by andygittings** [comment]
Still having the same issue with dgj223 and dgj419... :-(

As an example try:

amg17-mxs13@dgj223:~$ git clone https://github.com/andygittings/arc-python-hpc
Cloning into 'arc-python-hpc'...
fatal: unable to access 'https://github.com/andygittings/arc-python-hpc/': Received HTTP code 503 from proxy after CONNECT


Andy

**Comment by LiamATOS** [comment]
can we try again please from the login nodes

if it doesn't work, force unsetting the proxy env variables using - 

unset http{,s}_proxy

Thanks

Liam

**Comment by angad9** [comment]
Still not working (even after unsetting the proxy vars)

**Comment by LiamATOS** [comment]
can you provide me the full output please as it's working for me

Thanks

Liam

**Comment by angad9** [comment]
axs34-ttl03@dgj223:~$ ping www.google.com
PING www.google.com (108.177.126.147) 56(84) bytes of data.
^C
--- www.google.com ping statistics ---
6 packets transmitted, 0 received, 100% packet loss, time 5039ms

**Comment by LiamATOS** [comment]
Sorry, we don't allow all traffic out for security reasons.

Can you try git commands please, I tested and was able to clone from github.com

Thanks

**Comment by andygittings** [comment]
I can confirm both my git clones now work with the environment proxy variable unset.

Still seeing errors with pip install <package name> --user

This kind of thing:

Retrying (Retry(total=4, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f4d548d1a50>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/absl/
  Retrying (Retry(total=3, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f4d548d1110>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/absl/
  Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f4d548d1050>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/absl/
  Retrying (Retry(total=1, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f4d548d1350>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/absl/
  Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f4d548d13d0>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/abs


Andy

**Comment by andygittings** [comment]
amg17-mxs13@dgj223:~$ pip install pydicom --user
Collecting pydicom
  Retrying (Retry(total=4, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fa0b3c0de90>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/pydicom/
  Retrying (Retry(total=3, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fa0b3c0d410>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/pydicom/
  Retrying (Retry(total=2, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fa0b3c0d650>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/pydicom/
  Retrying (Retry(total=1, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fa0b3c0d7d0>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/pydicom/
  Retrying (Retry(total=0, connect=None, read=None, redirect=None)) after connection broken by 'NewConnectionError('<pip._vendor.requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fa0b3c0d910>: Failed to establish a new connection: [Errno 101] Network is unreachable',)': /simple/pydicom/
  Could not find a version that satisfies the requirement pydicom (from versions: )
No matching distribution found for pydicom

**Comment by angad9** [comment]
I am being able to clone from git as well. However, I can't seem to be able to install packages using conda: 

axs34-ttl03@dgj223:~$ conda install pytorch torchvision -c pytorch
Solving environment: failed

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.continuum.io/pkgs/pro/linux-64/repodata.json.bz2>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
ConnectionError(MaxRetryError("HTTPSConnectionPool(host='repo.continuum.io', port=443): Max retries exceeded with url: /pkgs/pro/linux-64/repodata.json.bz2 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f6f784d0358>: Failed to establish a new connection: [Errno 101] Network is unreachable',))",),)

**Comment by angad9** [comment]
Could you please help with the same?

**Comment by andygittings** [comment]
I think pip and conda are being affected by the same issue...

Andy

**Comment by syoffe** [comment]
I would also be interested in this issue being addressed, as we too cannot install or update packages using pip or conda.

Thank you for any help or advice you are able to offer,
Sam

**Comment by willprice** [comment]
I recall the `http_proxy` and `https_proxy` env vars being set previously, but when I check with `env | grep -i proxy` they are no longer present when I login, I notice they have now been commented out in `/etc/environment` on dgj223.

Interestingly I still have some form of external connectivity as I clone git repos, e.g. `git clone https://github.com/tmux/tmux.git`

...

I think this is probably some sort of firewall issue as 

```
$ wget github.com

URL transformed to HTTPS due to an HSTS policy
--2018-05-14 09:14:42--  https://www.github.com/
Resolving www.github.com (www.github.com)... 192.30.253.112, 192.30.253.113
Connecting to www.github.com (www.github.com)|192.30.253.112|:443... connected.
HTTP request sent, awaiting response... 301 Moved Permanently
Location: https://github.com/ [following]
--2018-05-14 09:14:42--  https://github.com/
Resolving github.com (github.com)... 192.30.253.113, 192.30.253.112
Connecting to github.com (github.com)|192.30.253.113|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘index.html’

index.html                                [ <=>                                                                   ]  53.41K   324KB/s    in 0.2s

2018-05-14 09:14:44 (324 KB/s) - ‘index.html’ saved [54690]


$ wget www.google.com
--2018-05-14 09:15:28--  http://www.google.com/
Resolving www.google.com (www.google.com)... 108.177.126.104, 108.177.126.103, 108.177.126.147, ...
Connecting to www.google.com (www.google.com)|108.177.126.104|:80... ^C
```

So I'm guessing if I try and clone from a non-gh remote then that'll fail as well, the same reason pip and conda fail to query their respective repositories.

Ping me if you need any troubleshooting on the user end of things :)

**Comment by LiamATOS** [comment]
Yes you are correct, only certain things are allowed through the firewall for security reasons.

github.com

Can you provide a list of things you need please and I can put the request in?

Thanks

Liam

**Comment by we-taper** [comment]
Besides the popular git repository, I would hope you made the UCL gitlab https://gitlab.cs.ucl.ac.uk/ whitelisted (if I understand it right).

Thanks.

**Comment by willprice** [comment]
Hi Liam,

- `http[s]://*.bitbucket.org`
- `http[s]://*.singularity-hub.org
- `http[s]://*.anaconda.org`
- `http[s]://*.conda-forge.org`
- `http[s]://python.org`
- `http[s]://*.pypi.org`
- `http[s]://*.pythonhosted.org`

Could you explain a bit more about why the decision was made to blacklist by default?

**Comment by LiamATOS** [comment]
Requests have been made for all of the above, will let you know once its been done.

I will ask for further clarification from STFC on the change in policy as all I received was for security reasons.

Thanks

Liam

**Comment by willprice** [comment]
Hi Liam,

Thanks very much for the quick response and forwarding of the question; it's much appreciated.

**Comment by LiamATOS** [comment]
Ok, after many complaints, STFC have decided to allow everything out on 443 (https) from both of the login nodes.

I am hoping this will resolve most of the issues, I will confirm once its been done by the network team.

Thanks

Liam

**Comment by syoffe** [comment]
Hello,
I was just wondering if there has been any update on this? I am still unable to `conda update -n root conda` and `ping` also fails. This is quite a long-standing issue now and we're rapidly approaching a grant proposal deadline which we need some simulation results for!
Thanks for your help,
Sam

**Comment by willprice** [comment]
@LiamATOS,
Any news? This still remains quite a blocking issue for many of us in our lab.
Many thanks,

**Comment by LiamATOS** [comment]
Hi, Can you try again, I have tested and https traffic from both the login node was working

Thanks

Liam

**Comment by syoffe** [comment]
I logged out and back in (connected to jade.hartree.stfc.ac.uk which has brought me to node dgj419. I tried `conda update` once again to be presented with this error again:
```Solving environment: failed

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

`$ miniconda3/bin/conda update -n root conda`

  environment variables:
                 CIO_TEST=<not set>
               CONDA_ROOT=/jmain01/home/JAD024/sry01/sry31-sry01/miniconda3
               MODULEPATH=/jmain01/apps/modules:/etc/environment-modules/modules:/usr/share/modu
                          les/versions:/usr/Modules/$MODULE_VERSION/modulefiles:/usr/share/modul
                          es/modulefiles
                     PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/game
                          s:/usr/local/games:.:/jmain01/home/JAD024/sry01/sry31-sry01/bin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : None
       user config file : /jmain01/home/JAD024/sry01/sry31-sry01/.condarc
 populated config files : 
          conda version : 4.5.1
    conda-build version : not installed
         python version : 3.6.5.final.0
       base environment : /jmain01/home/JAD024/sry01/sry31-sry01/miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/pro/linux-64
                          https://repo.anaconda.com/pkgs/pro/noarch
          package cache : /jmain01/home/JAD024/sry01/sry31-sry01/miniconda3/pkgs
                          /jmain01/home/JAD024/sry01/sry31-sry01/.conda/pkgs
       envs directories : /jmain01/home/JAD024/sry01/sry31-sry01/miniconda3/envs
                          /jmain01/home/JAD024/sry01/sry31-sry01/.conda/envs
               platform : linux-64
             user-agent : conda/4.5.1 requests/2.18.4 CPython/3.6.5 Linux/4.4.0-112-generic ubuntu/16.04 glibc/2.23
                UID:GID : 6181:30809
             netrc file : None
           offline mode : False


V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V V

CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.anaconda.com/pkgs/main/linux-64/repodata.json.bz2>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.

If your current network has https://www.anaconda.com blocked, please file
a support request with your network engineering team.

ConnectionError(MaxRetryError("HTTPSConnectionPool(host='repo.anaconda.com', port=443): Max retries exceeded with url: /pkgs/main/linux-64/repodata.json.bz2 (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f924ffd8278>: Failed to establish a new connection: [Errno 101] Network is unreachable',))",),)```

**Comment by willprice** [comment]
@LiamATOS I too have the same issue as @syoffe. Nor can I wget google.com:
```
$ wget https://google.com
--2018-06-05 15:04:36--  https://google.com/
Resolving google.com (google.com)... 216.58.208.174, 2a00:1450:4009:808::200e
Connecting to google.com (google.com)|216.58.208.174|:443... ^C
```
It just hangs.

Are the compute nodes not getting https access? It'd be helpful for utilising experimental logging frameworks like [Comet.ml](https://www.comet.ml/)

**Comment by willprice** [comment]
I see that the `xcat` proxy server is back up and running but the environment variables are still commented out in `/etc/environment`

Adding `_proxy="http://xcat:3128"; export http_proxy="$_proxy"; export https_proxy="$_proxy"` to one's `~/.bashrc` resolves the issue.

**Comment by syoffe** [comment]
Thank you, @willprice! That's working for me now, too!

**Comment by llyhec** [comment]
I am still having problems with this: 


>>conda create -n vtest1 python=3.5
Fetching package metadata ...
CondaHTTPError: HTTP 000 CONNECTION FAILED for url <https://repo.continuum.io/pkgs/main/linux-64/repodata.json.bz2>
Elapsed: -

An HTTP error occurred when trying to retrieve this URL.
HTTP errors are often intermittent, and a simple retry will get you on your way.
ConnectionError(MaxRetryError("HTTPSConnectionPool(host='repo.continuum.io', port=443): Max retries exceeded with url: /pkgs/main/linux-64/repodata.json.bz2 (Ca                             used by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7ff2a6433470>: Failed to establish a new connection: [Errno -3] Temporary fa                             ilure in name resolution',))",),)


Could you please help with this?

**Comment by syoffe** [comment]
Hi @llyhec,

Did you follow the instructions in @willprice's post? Your exact command worked for me.

> Adding `_proxy="http://xcat:3128"; export http_proxy="$_proxy"; export https_proxy="$_proxy"` to one's `~/.bashrc`

Also, have you installed your own `conda` distribution or are you using one provided by JADE? I have my own `miniconda` installation, and have not used the one provided by them.

Good luck,
Sam

**Comment by llyhec** [comment]
Thanks Sam, yes I had the proxy in my bashrc file. I restarted everything and now it seems to work. I am using the one provided by JADE - I will install my own miniconda.

**Comment by syoffe** [comment]
Yeah after making changes to your `.bashrc` file you either need to do `source ~/.bashrc` or log out and back in. Glad it's working now!

**Comment by rupertnash** [comment]
Please also add *.nektar.info
